themes/default/content/blog/iac-recommended-practices-structuring-pulumi-projects/index.md

```diff
@@ -0,0 +1,98 @@
+
+Finally, there is the consideration of what happens in the event of human error, and limiting the so-called "blast radius." With a single project, _all_ the resources in that project are within the blast radius---all of them could potentially be affected in the event of human error. Using separate Pulumi projects can limit the blast radius. It's not something that should drive the decision all by itself, but it _is_ a factor to be considered.
+
+_In general, most customers end up using multiple Pulumi projects per use case._ As you've seen, though, there are a number of factors that influence that decision and will affect how many projects end up in use.
+
+## Moving forward with Zephyr
+
+Zephyr had a number of factors in play, not the least of which was the organizational change resulting from their growth and success. In the end, Zephyr decided to restructure their use of Pulumi for the online retail store into three separate projects:
+
+* An infrastructure project: This project handled the AWS VPC and related infrastructure resources.
+* A platform project: This project managed the Kubernetes cluster(s).
+* An application project: This project handled the actual online retail store itself.
+
+{{% notes %}}
+Note that today, Pulumi has no supported method to move resources from one project to another, so project restructuring will involve creating new infrastructure and migrating over to it.
+{{% /notes %}}
+
+In [the first Zephyr post](/blog/iac-recommended-practices-code-organization-and-stacks/), you read about some of the options for where to store Pulumi code relative to the applications it supports. At that time, Zephyr had decided to go with what's called a monorepo approach, meaning that the Pulumi code was stored in the same repository as the application code. In conjunction with the project reorganization, Zephyr also decided to change how their code was organized:
+
+* Code for the infrastructure project moved to a separate "zephyr-infra" repository
+* A new "zephyr-k8s" repository housed the code for the platform project
+* The code for the online retail store application itself remained in the same "zephyr-app" repository
+
+Note that relocating code (to a different GitHub repository or different filesystem location within a repository) generally has no impact on Pulumi---keeping in mind, of course, that filesystem paths and such may need to be adjusted in code to account for any such changes.
+
+These repositories are available for you to review on GitHub; use the branch/tag selector to find the `multi-project` tab in each repository to see the state of the code as of this blog post.
```

Comment:
do you want to just hyperlink to it here?

Reply:
I will hyperlink to the repos here, but the repos don't exist yet. :-)

Reply:
I will add the hyperlinks after the repos are created but before the post goes live.

Reply:
Agree a table or something with links to each would be nice here.

Reply:
What do you think about maybe showing (or at least explaining) how these are (or will need to be) deployed now?

Reply:
I've added hyperlinks to the repositories, and added a note about checking the instructions in each repo to see how to deploy the online store now that it is in multiple projects.